compiler/src/main/scala/org/scalajs/core/compiler/PrepJSExports.scala

```diff
@@ -100,48 +100,63 @@ trait PrepJSExports { this: PrepJSInterop =>
     }
   }
 
-  /** Checks and registers module exports on the symbol */
-  def registerModuleExports(sym: Symbol): Unit = {
-    assert(sym.isModuleClass, "Expected module class")
-    registerClassOrModuleExportsInternal(sym)
-  }
+  /** Check and (potentially) register a class or module for export.
+   *
+   *  Note that Scala classes are never registered for export, their
+   *  constructors are.
+   */
+  def registerClassOrModuleExports(sym: Symbol): Unit = {
+    val exports = exportsOf(sym)
+    def isScalaClass = !sym.isModuleClass && !isJSAny(sym)
 
-  /** Checks and registers class exports on the symbol. */
-  def registerClassExports(sym: Symbol): Unit = {
-    assert(!sym.isModuleClass && !sym.hasAnnotation(JSNativeAnnotation),
-        "Expected a Scala.js-defined JS class")
-    registerClassOrModuleExportsInternal(sym)
+    if (exports.nonEmpty && checkClassOrModuleExports(sym, exports.head.pos) &&
+        !isScalaClass) {
```

Comment:
Since `checkClassOrModuleExports` is called through `genExportMember` for constructors of Scala classes, shouldn't the test `!isScalaClass` come before the test `checkClassOrModuleExports`? Currently it seems to me that there is a duplicate check.